### PR TITLE
fixed requirements and typings in train.py in task1

### PR DESCRIPTION
**Pull request by VladPyzh**
requirements.txt:

Changed `torchvision` version to `0.17.0`
Added `matplotlib`

train.py:

Changed typings `torch.nn._Loss` -> ` torch.nn.modules.loss` and `torch.optim.optimizer.Optimizer` -> `torch.optim.Optimizer`

### CODE REVIEW

**Comment by mac-mvak** [comment]
Мне кажется, учитывая, что программа 
```
import torch

model = torch.nn.Linear(5, 5)
optim = torch.optim.Adam(model.parameters())
loss = torch.nn.MSELoss()

def func(a : torch.nn.modules.loss):
    print('aaa')



print(isinstance(loss, torch.nn.modules.loss))
```
падает с тем, что `torch.nn.modules.loss` это не тип, то лучше мой коммит принять.

**Comment by VladPyzh** [comment]
пофиксил на `criterion: torch.nn.modules.loss._Loss`

**Review by mryab — APPROVED** [review]
Thanks for the contribution!
